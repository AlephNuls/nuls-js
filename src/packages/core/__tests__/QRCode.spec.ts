import { QRCode } from '@/index';

describe('creating QR codes', () =>
{
	test('an address', async () =>
	{
		const code = await QRCode.create('Nse1TYHc6Rxs84iimrnygSF2kqrUAQM6');

		expect(code).toMatchSnapshot();
	});

	test('an address with an amount', async () =>
	{
		const code = await QRCode.create('Nse1TYHc6Rxs84iimrnygSF2kqrUAQM6', { amount: 10 });

		expect(code).toMatchSnapshot();
	});
});
