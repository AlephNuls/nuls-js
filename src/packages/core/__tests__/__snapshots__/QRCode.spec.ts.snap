// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating QR codes an address 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAALQAAAC0CAYAAAA9zQYyAAAAAklEQVR4AewaftIAAAdrSURBVO3BQY4kNhLAQFLo/3+Z62OeBBSqZ+wVMsL+wVqPOKz1kMNaDzms9ZDDWg85rPWQw1oPOaz1kMNaDzms9ZDDWg85rPWQw1oPOaz1kMNaDzms9ZAfvqTyN1VMKlPFN1SmiknlGxV/k8pU8QmVv6niG4e1HnJY6yGHtR7ywy+r+E0qn1C5qZhUPlHxCZVJ5abiGypTxaQyVXyi4jep/KbDWg85rPWQw1oP+eEPU/lExTcq/k0qNxWTyqQyVUwqNxV/k8onKv6kw1oPOaz1kMNaD/nh/1zFpDJV3FRMKpPKVDGp3FRMKjcVNxWTyqRyU/Gyw1oPOaz1kMNaD/nhcSpTxaRyU/GJikllqviTKj6hMlX8Pzus9ZDDWg85rPWQH/6wij9J5aZiUrmpmFRuKm4qJpWbiknlpuITKr+p4r/ksNZDDms95LDWQ374ZSr/popJZaqYVD5RMalMFZPKVDGpfKJiUpkqJpWpYlL5hMp/2WGthxzWeshhrYf88KWK/7KKSWWqmFR+U8Wk8omKSeU3VdxU/D85rPWQw1oPOaz1kB/+MJVPVEwqU8UnVKaKSWWq+IbKJyomlaliqvhExaQyVUwqn6i4UflExTcOaz3ksNZDDms95IcvqXyiYlKZVKaKb1R8Q2WqmCq+oXKjclNxo3KjMlVMKlPFjcpUMalMFb/psNZDDms95LDWQ374ZRWTyqQyVdyofEJlqphUpoqbihuVm4pJ5abiT6r4TSpTxSdUpopvHNZ6yGGthxzWesgPf1jFpDKp3FTcqHyiYlL5RMU3KiaVSWWqmFSmim+o3FRMKlPFf8lhrYcc1nrIYa2H/PClij9J5aZiUvk3VUwqU8VUcaNyo3JT8YmKSWWqmFSmik9U/KbDWg85rPWQw1oP+eFLKp+ouFGZKiaVT6h8o2JSmSo+oTJVfKPiRmWqmComlaliUpkqblSmikllqvjGYa2HHNZ6yGGth9g/+INUbiomlZuKG5W/qWJSmSpuVKaKG5WbiknlpuJG5aZiUvlGxTcOaz3ksNZDDms95IdfpjJVTCqfqLhRuamYVD5RMal8QuWm4hMVk8pNxY3KVDFV3KhMFTcqf9JhrYcc1nrIYa2H/PCXVdxUTCpTxb+pYlKZKiaVqWJSuamYVKaKSeUbKp+ouFGZKv6kw1oPOaz1kMNaD/nhSyp/UsWkclMxqUwVNyqfqJhUPlHxDZWbik+oTBWTyo3Kv+mw1kMOaz3ksNZD7B/8RSo3FZPKTcUnVKaK36QyVdyoTBWTyjcqJpVvVEwqNxWTyk3FNw5rPeSw1kMOaz3kh1+m8omKm4pJ5UZlqpgqblQ+UXGj8psqPqEyVdyoTBWTylTxX3JY6yGHtR5yWOshP/yyihuVSeXfpHJTMalMKn+TylQxqdyofELlRuW/5LDWQw5rPeSw1kN++JLKn1TxDZWbihuVqeJGZaqYVKaKm4pJ5RsVk8o3KiaVqeJG5Tcd1nrIYa2HHNZ6iP2DL6hMFTcq36iYVKaKSeUTFTcqNxWTyn9ZxaQyVXxC5RsV3zis9ZDDWg85rPWQH75UcaPyjYpJZaq4qZhUblRuKv6fVNyoTBU3KjcVNyp/0mGthxzWeshhrYf88MtUpooblaliUpkqPqHyiYpPqNxU3KjcVNyoTBWTylRxozJVfELl33RY6yGHtR5yWOsh9g++oPKJikllqrhRmSo+oTJV3KjcVEwqU8WkMlVMKjcV31CZKm5Upooblanibzqs9ZDDWg85rPWQH35ZxaTyDZWp4hMqNypTxVRxozJVTCpTxaTyCZWbik+oTBVTxaQyVUwVNyo3Fd84rPWQw1oPOaz1kB/+sIpJZaqYVKaKSWWquKn4hspNxaTymyomlaniRuWm4hsqU8W/6bDWQw5rPeSw1kPsH3xBZaqYVG4qblQ+UTGp3FT8SSo3FTcqNxU3Kv+mir/psNZDDms95LDWQ+wf/B9Tuam4UZkqblRuKiaVqWJSmSomlaniRuUbFZ9QmSpuVG4qvnFY6yGHtR5yWOshP3xJ5W+qmCpuVKaKqWJSmSpuKm4qPqEyVdyoTBU3Kp9QmSr+yw5rPeSw1kMOaz3kh19W8ZtUfpPKVDFVTCpTxaQyVdyoTBU3Kt9QmSomlZuK31Qxqfymw1oPOaz1kMNaD/nhD1P5RMUnVG4qPqFyo3KjMlXcqEwVNxU3KjcqNyrfUPk3HdZ6yGGthxzWesgPj6m4UflNFZPKpPInqUwVn1D5kyr+psNaDzms9ZDDWg/54TEqNxU3Kt+omFQ+oTJV3FR8QmWquFG5qbhRmSomlaniG4e1HnJY6yGHtR7ywx9W8SdV3KhMKlPFTcWk8idVfEJlqphUpopJZaqYKm5Ubir+psNaDzms9ZDDWg/54Zep/JtUbiomlaniGypTxaRyo3JTMVVMKp+ouFH5RMWk8jcd1nrIYa2HHNZ6iP2DtR5xWOshh7UecljrIYe1HnJY6yGHtR5yWOshh7UecljrIYe1HnJY6yGHtR5yWOshh7UecljrIf8DI/rGmq7M1u4AAAAASUVORK5CYII="`;

exports[`creating QR codes an address with an amount 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAMQAAADECAYAAADApo5rAAAAAklEQVR4AewaftIAAAjMSURBVO3BQY4kx7IgQdVA3f/KOo2/cNjKgUBmNck3JmJ/sNb6Pw9rreNhrXU8rLWOh7XW8bDWOh7WWsfDWut4WGsdD2ut42GtdTystY6HtdbxsNY6HtZax8Na6/jhQyp/U8WNyicq3lC5qZhUpooblZuKSeWmYlKZKiaVNyomlb+p4hMPa63jYa11PKy1jh++rOKbVL6p4kZlqripmFQ+oTJVTCpvVNxUTCpTxTdVfJPKNz2stY6HtdbxsNY6fvhlKm9UfKJiUplUpooblaniDZWpYlKZKt5QmSpuVKaKqWJSmSomlaniDZU3Kn7Tw1rreFhrHQ9rreOH/3EVNyqfUPk3U5kqJpWp4kZlqvhf8rDWOh7WWsfDWuv44T9O5aZiUpkqblRuKm5UJpWp4kZlqrhRuVGZKiaV/589rLWOh7XW8bDWOn74ZRW/qeJG5ZtU3qiYVCaVm4pJ5RMVk8pUcaPyTRX/Jg9rreNhrXU8rLWOH75M5W9SmSpuKiaVqeKmYlKZKiaVqWJSmSomlaliUpkqJpVPqEwVk8pUcaPyb/aw1joe1lrHw1rr+OFDFf8mKlPFTcU3qbxRcVPxTSo3Kjcqb1T8lzystY6HtdbxsNY6fvhlKjcVNypvVEwqNxWTyr+JyhsqNxVvqLyhclMxqUwV/6SHtdbxsNY6HtZah/3BB1S+qeINlZuKSeWm4kZlqrhRuan4hMobFTcqU8WkMlW8ofKJikllqvjEw1rreFhrHQ9rrcP+4AMqU8WkMlXcqNxUvKEyVUwq31TxTSpTxaQyVbyhMlVMKlPF/7KHtdbxsNY6HtZaxw8fqvhNFZPKVDGp3KhMFZ9QuVG5qZhUpopJ5UZlqphU3qiYVD5R8QmVm4pPPKy1joe11vGw1jrsDz6gMlVMKlPFGyo3FZ9QeaNiUrmp+ITKTcWk8k0Vb6hMFW+o3FRMKlPFJx7WWsfDWut4WGsdP3yoYlK5UXmjYlK5UbmpuKmYVCaVqeKbVG4qbiomlaniRmVSmSp+U8WNym96WGsdD2ut42GtddgffEBlqrhR+UTFpPI3VbyhclNxo3JT8b9EZaqYVG4qPvGw1joe1lrHw1rr+OFDFZPKVDFVTCpTxRsVk8pUMancVEwqb6jcVEwqU8U3qdxUvKHyT6qYVL7pYa11PKy1joe11vHDh1RuVKaKqWJSeaNiqvhNKjcVNyo3KlPFGyo3FW+o3FRMKlPFjcq/ycNa63hYax0Pa63jh79MZaqYKm5UblT+TVSmijdUpopJZaqYVCaVqWJSmSreqJhUpoqpYlK5UZkqvulhrXU8rLWOh7XWYX/wD1L5mypuVG4qPqEyVUwqU8WkMlW8ofJGxRsqb1RMKlPF3/Sw1joe1lrHw1rr+OGXqdxUTCpTxY3KVHGj8kbFpDJVTCo3FTcVn1C5qZhUpooblanipmJSeUNlqphUpopPPKy1joe11vGw1jp++JDKVPGGylQxqdxU3KhMFZ+ouKmYVCaVm4o3VG4qPqFyo3KjMlXcVEwqf9PDWut4WGsdD2ut44cvU5kq3lCZKiaVSWWquFH5RMWkMlVMFTcqn6iYVCaVfxOVqeKmYlL5TQ9rreNhrXU8rLUO+4MvUvk3qZhUpopJZaqYVKaKSWWqmFSmihuVqeKbVN6o+ITKN1V808Na63hYax0Pa63jhy+reENlqrhReUNlqnhDZar4JpWp4g2Vm4pJ5abiRuVvqphUftPDWut4WGsdD2ut44cPqdxUTCpTxY3KVDGpTBWTyjepTBVTxaQyVUwqk8pUcVMxqbxR8YmKT6jcqEwVk8pU8YmHtdbxsNY6HtZah/3BB1RuKv7LVN6ouFGZKt5QeaNiUrmpmFRuKr5J5RMVn3hYax0Pa63jYa112B98QOWm4g2Vm4pJZaqYVG4qJpWbiknlpuJG5Zsq3lCZKm5U3qiYVKaKT6hMFZ94WGsdD2ut42GtdfzwZRWTyk3FVHGj8kbFpDKp/CaVqWKqeEPlDZWbik9UTCqTyo3KTcWkMlV808Na63hYax0Pa63jh19WMalMKp+omFSmijcqJpWbikllqphUpopJ5aZiUrmpmFTeUJkqPlExqUwV/6SHtdbxsNY6HtZah/3BF6lMFZPKTcVvUpkqJpWpYlJ5o2JSeaPib1KZKiaVqeJGZaq4UZkq/qaHtdbxsNY6HtZah/3BB1SmiknlmyomlaliUpkqPqEyVdyoTBVvqEwVNyo3Fd+kMlVMKlPFpPJGxW96WGsdD2ut42Gtddgf/INUpoo3VG4qJpU3Km5UbiomlaliUpkqJpWp4kblmyomlaliUnmjYlKZKiaVqeITD2ut42GtdTystY4fvkzlpuINlZuKG5WbiknlExWTylQxqdyoTBU3Km9UTCo3Kjcq/2UPa63jYa11PKy1DvuD/zCVm4oblaliUpkqJpU3KiaVb6qYVKaKSWWqmFTeqHhDZar4Jz2stY6HtdbxsNY6fviQyt9UcVMxqUwVb1RMKm9U3FRMKlPF36QyVUwqb6hMFZ9QmSq+6WGtdTystY6Htdbxw5dVfJPKTcWkMlVMKm+o/CaVN1SmipuKSWWquFH5RMU3Vfymh7XW8bDWOh7WWscPv0zljYpPVEwqU8WkMqn8l6ncVEwqU8VNxaQyqXxCZar4mx7WWsfDWut4WGsdP/zHqXxTxaRyUzGpvFExqUwqU8VNxW9SeaPiRmWqmFSmikllqvjEw1rreFhrHQ9rreOH/zEVNypvVEwqk8pUMalMFTcVk8obKlPFTcVNxY3KGypTxaQyVfxND2ut42GtdTystY4fflnFb6r4TSrfpDJVTCqfqHhDZar4JpWpYlKZKm5UpopvelhrHQ9rreNhrXX88GUq/0sqPlExqUwqU8UnVG4qpopJ5ZsqPqHyNz2stY6HtdbxsNY67A/WWv/nYa11PKy1joe11vGw1joe1lrHw1rreFhrHQ9rreNhrXU8rLWOh7XW8bDWOh7WWsfDWut4WGsd/w/vNJvdgPL+3AAAAABJRU5ErkJggg=="`;
